django/cantusdb_project/main_app/views/chant.py

```diff
@@ -1139,45 +1107,84 @@ def get_suggested_chants(self):
+                    the database
 
             Returns:
-                dict: dictionary containing data for a specific chant, along with a count
-                of how many times it appears after instances of the other particular chant.
+                dict: a dictionary with data about a specific Cantus ID, including:
+                    - a canonical fulltext (str)
+                    - an incipit (str)
+                    - a genre (str)
+                    - the ID of that genre in Cantus Database (int)
+                    - how many times chants with this Cantus ID follow chants with a
+                        Cantus ID of 123456 (int)
             """
-            sugg_chant_cantus_id, sugg_chant_count = suggested_chant
-            # search Cantus Index
-            response = requests.get(
-                f"https://cantusindex.org/json-cid/{sugg_chant_cantus_id}",
-                timeout=5,
-            )
-            assert response.status_code == 200
-            # parse the json export to a dict
-            # can't use response.json() because of the BOM at the beginning of json export
-            chant_dict = json.loads(response.text[2:])[0]
-            # add number of occurence to the dict, so that we can display it easily
-            chant_dict["count"] = sugg_chant_count
-            # figure out the id of the genre of the chant, to easily populate the Genre selector
-            genre_name = chant_dict["genre"]
-            genre_id = Genre.objects.get(name=genre_name).id
-            chant_dict["genre_id"] = genre_id
-            return chant_dict
+            try:
+                response = requests.get(
+                    f"https://cantusindex.org/json-cid/{cantus_id}",
+                    timeout=5,
+                )
+                assert response.status_code == 200
+                # we can't use response.json() because of the BOM at the beginning of json export
+                cid_dict: dict = json.loads(response.text[2:])[0]
+                # add number of occurence to the dict, so that we can display it easily
+                cid_dict["count"] = count
+                # figure out the id of the genre of the chant, to easily populate the Genre selector
+                genre_name = cid_dict["genre"]
+                genre_id = Genre.objects.get(name=genre_name).id
+                cid_dict["genre_id"] = genre_id
+            except (
+                AssertionError,
+                SSLError,
+                Timeout,
+            ) as exc:
+                print(  # eventually, we should log this rather than printing it to the console
+                    "Encountered an error in",
+                    "ChantCreateView.get_suggested_chants.make_suggested_chant_dict",
+                    f"while making a request to https://cantusindex.org/json-cid/{cantus_id}:",
+                    exc,
+                )
+                cid_dict = {}
+            except (
```

Comment:
The JSON Loading should be contained in it's own try/except. You should be very minimal in what goes in the body of the `try` and not do too many things in one try statement.

"Additionally, for all try/except clauses, limit the try clause to the absolute minimum amount of code necessary. Again, this avoids masking bugs"

https://peps.python.org/pep-0008/